trustfall_core/src/ir/mod.rs

```diff
@@ -282,6 +299,15 @@ impl Argument {
     }
 }
 
+/// Operations that can be made in the graph.
+///
+/// In GraphQL, this can correspond to the `op` argument in `@filter`,
+/// for example in the following:
+/// ```graphql
+/// query Student {
+///     name @filter(op: "has_substring", values: ["John"])
+/// }
```

Comment:
```suggestion
/// In a Trustfall query, the `@filter` directive produces `Operation` values:
/// ```graphql
/// name @filter(op: "=", values: ["$input"])
/// ```
/// would produce the `Operation::Equals` variant, for example.
```
Because the other query wasn't actually valid:
- It has a property field on the root node, which in Trustfall should produce a schema error when attempting to parse it. (Not sure this is tested! If not, probably a great entry point for contributing some code, if you're interested, since you almost certainly know more GraphQL than I do!)
- `@filter` can't take literal values at the moment, only `"$foo"` runtime arguments or `"%foo"` tagged arguments (see the `@tag` directive, it "captures" the tagged value and allows subsequent filters to refer back to it).

I felt that a fully-valid query would be too complex (at least two more lines), so I went with an excerpt. What do you think?

Reply:
I agree, this is better. I will look into the property field on the root node!